File: Security/src/AspDotNet4/CloudFoundrySingleSignon/AuthorizationManager.cs
using System.Linq;
using System.Security.Claims;

namespace CloudFoundrySingleSignon
{
    public class AuthorizationManager : ClaimsAuthorizationManager
    {
        public override bool CheckAccess(AuthorizationContext context)
        {
            var requiredClaims = context.Action.Select(v => v.Value);
            var possessedClaims = context.Principal.Claims.Select(v => v.Value);

            var claim = ((ClaimsIdentity)context.Principal.Identity).FindFirst("testgroup");

            foreach (var c in requiredClaims)
            {
                if (possessedClaims.Contains(c))
                {
                    return true;
                }
            }

            return false;
        }
    }
}

File: Security/src/AspDotNet4/CloudFoundrySingleSignon-VB/AuthorizationManager.vb
Imports System.Security.Claims

Namespace CloudFoundrySingleSignon
    Public Class AuthorizationManager
        Inherits ClaimsAuthorizationManager

        Public Overrides Function CheckAccess(ByVal context As AuthorizationContext) As Boolean
            Dim requiredClaims = context.Action.[Select](Function(v) v.Value)
            Dim possessedClaims = context.Principal.Claims.[Select](Function(v) v.Value)
            Dim claim = (CType(context.Principal.Identity, ClaimsIdentity)).FindFirst("testgroup")

            For Each c In requiredClaims

                If possessedClaims.Contains(c) Then
                    Return True
                End If
            Next

            Return False
        End Function
    End Class
End Namespace
